msteams-platform/bots/bot-features.md

```diff
@@ -47,6 +48,105 @@ The [Bot Framework](https://dev.botframework.com/) is a rich SDK used to create
+>   * You can also use az-cli to do bot registration:
+>
+>     1. Sign into azure and set the subscription <br> 
+>           &nbsp; az cloud set –name  "AzureUSGovernment" <br> 
+>           &nbsp; az account set –name "`subscriptionname/id`".<br>
+>     1. Create app registration  
+>           &nbsp; az ad app create --display-name "`name`" <br> 
+>           &nbsp; --password "`password`" --available-to-other-tenants.<br> 
+>           Your app id would be created here.<br>
+>     1. Create bot resource <br>
+>           &nbsp; az bot create –resource-group "`resource-group`"<br>
+>           &nbsp; --appid "`appid`"<br>
+>           &nbsp; --name "`botid`"<br>
+>           &nbsp; --kind "registration".<br>
+>
+> </details>
+
+For GCCH environment, you need to register a bot using [Azure Government portal](https://portal.azure.us).
+
+:::image type="content" source="../assets/videos/abs-bot.gif" alt-text="Azure Government portal":::
+<br>
+<br>
+The following changes are needed within the bot for GCC-High environment:
+<br>
+<br>
+<details>
+<summary><b>Configuration changes</b></summary>
+
+As the bot registration occurs in Azure Government portal, ensure to update the bot configurations to connect to Azure govermnet instances. Following are the configuration details:
+
+| Configuration Name | Value |
+|----|----|
+| ChannelService | `https://botframework.azure.us` |
+| OAuthUrl | `https://tokengcch.botframework.azure.us` |
+| ToChannelFromBotLoginUrl | `https://login.microsoftonline.us/MicrosoftServices.onmicrosoft.us` |
+| ToChannelFromBotOAuthScope | `https://api.botframework.us` |
+| ToBotFromChannelTokenIssuer | `https://api.botframework.us`  |
+| BotOpenIdMetadata | `https://login.botframework.azure.us/v1/.well-known/openidconfiguration` |
+
+</details>
+<br>
+<details>
+<summary><b>Update to appsettings.json & startup.cs</b></summary>
+
+1. **Update appsettings.json:**
+
+    * Set `ConnectionName` to the name of the OAuth connection setting you added to your bot.
+
+    * Set `MicrosoftAppId` and `MicrosoftAppPassword` to your bot's app ID and app secret.
+    
+    Depending on the characters in your bot secret, you may need to XML escape the password. For example, any ampersands (&) need to be encoded as `&amp;`.
+
+    ```json
+    {
+      "MicrosoftAppType": "",
+      "MicrosoftAppId": "",
+      "MicrosoftAppPassword": "",
+      "MicrosoftAppTenantId": "",
+      "ConnectionName": ""
+    }
+    ```
+2. **Update Startup.cs:**
+
+    To use OAuth in *non-public Azure clouds*, like the government cloud, or in bots with data-residency, you must add the following code in the **Startup.cs** file.
+    
+    ```csharp
+    string uri = "<uri-to-use>";
+    MicrosoftAppCredentials.TrustServiceUrl(uri);
```

Comment:
This is deprecated now.

Reply:
Should this be replaced with updated information or removed?